cpp/harvest/core.cc

```diff
@@ -553,6 +556,95 @@ void DataProviders(int argc, char **argv) {
 }
 
 
+void SplitDataProviderId(int argc, char **argv) {
+    if (argc != 4)
+        Usage();
+
+    std::cout << "Preparing data..." << std::endl;
+    const auto works(CORE::GetWorksFromFile(argv[2]));
+
+    std::vector<unsigned> list_of_data_provider_id;
+    std::map<std::string, unsigned> data_provider_id_counter;
+    unsigned total_record_counter(0), progress(0), step(0), display_next(0), closed_counter(0);
+    std::string item, outputFile, outputDir(argv[3]);
+
+
+    std::cout << "Processing " << works.size() << " records ..." << std::endl;
+    for (auto work : works) {
+        if (!work.getDataProviderIds().empty()) {
+            for (auto data_provider_id : work.getDataProviderIds()) {
+                item = std::to_string(data_provider_id);
+                outputFile = outputDir + item + ".json";
```

Comment:
are you absolute sure that outputDir has a trailing slash?

Reply:
I added a checker on wether it has a backslash or not.